cms/core/url_utils.py

```diff
@@ -60,3 +60,10 @@ def normalise_url(url: str) -> str:
     url = url.lower().rstrip("/")  # Treat URLs with and without trailing slashes as equivalent
     url = url.removeprefix("https://").removeprefix("www.")  # Normalise the URL
     return url
+
+
+def extract_url_path(url: str) -> str:
+    """Extracts and returns the path component of a URL."""
+    parsed_url = urlparse(url)
+    path = parsed_url.path.lower().rstrip("/")  # Treat paths with and without trailing slashes and caps as equivalent
```

Comment:
I get what this is aiming for, but I’m not convinced the function should be doing the lower-casing. Its job is simply to return the path as it appears in the URL. Once it starts modifying the value, especially around case, it opens up room for subtle bugs because URLs can be case-sensitive. If another part of the code relies on the original casing or uses the path for a redirect, this could cause unexpected behaviour. It feels cleaner for the caller to decide how the path should be normalised rather than baking that into this helper.